### ISSUE TITLE
Cloudflare

### ISSUE THREAD

**Issue by adamluk** [jawj/IKEv2-setup#23]
Hello,

Thanks for your work on this, very useful. 

Their is an issue when using this script with a domain which has its DNS hosted at Cloudflare. It is necessary to disable the proxy feature and use DNS only, otherwise certbot fails. 

Network interface: eth0
External IP: XXX
Warning: vpn.LLLL.net resolves to 104.28.23.28
104.28.22.28, not XXX
Either you are behind NAT, or something is wrong
Press [Return] to continue, or Ctrl-C to abort

Are you using Cloudflare?

If this was identified at this point as being an issue the option could even be given, so that:

certbot could be set to use the --standalone-supported-challenges http-01 flag

That's only one solution and I appreciate it may not be the best?

**Comment by jawj** [comment]
Are you saying if we switched the Let's Encrypt command to `--standalone-supported-challenges http-01` or `--preferred-challenges http` (as described at http://letsencrypt.readthedocs.io/en/latest/using.html#standalone), this would fix the problem with Cloudflare? If so, can you point me to documentation that explains why? I've no Cloudflare experience.

I can't foresee any problem with this for anyone else, so if that's the case it would be an easy change.

**Comment by adamluk** [comment]
I've spent some more time looking into this and although using the --preferred-challenges http flag gets around what I identified as the initial problem it is not a complete solution. In short I no longer feel that running a VPN over Cloudflare is a great idea. In fact they recommend against it. That being the case I'm happy for this issue to be closed. For the record I am using Cloudflare in DNS mode only, with the proxy function disabled - and all is functioning well.

Line 177 of the script could therefore be edited to:

certbot certonly --non-interactive --agree-tos --email $EMAIL --standalone --preferred-challenges tls-sni -d $VPNHOST

This then specifies that port 443 will be used.
